### DISCUSSION TITLE
Grid Down, Battery Alarm, and 24-hour sensors

### DISCUSSION THREAD

**Discussion by sfox38** [davidrapan/ha-solarman#286] [Q&A]
Got your awesome integration up and running with my SUN-10K-SG04LP3-EU and I think it's all working quite well. Now that it's been running a few days, I have a few issues/questions/suggestions:

I am noticing a lot of events from binary_sensor.inverter_grid. By a lot, I mean 20 to 30 times per day, at random times. Typically, it will indicate the grid has gone down, and stays down for just a minute or two, and all the other grid power sensors also go to zero during that time. The Solarman iOS app still shows the grid is up and providing power, however! And to the best of my knowledge, the grid did not actually go down and power was not cut. Well it did go down for real once, and the behavior of the integration was identical. I wish this sensor was more accurate so that I can use it to send alerts and turn certain devices on or off depending on the grid status. What actually triggers that sensor? What issues might I look out for regarding our seemingly now unstable grid power supply.

I like all the handy Total and Daily sensors, is there any chance of also adding built-in 24-hour sensors? I'd prefer to measure power over the past 24 hours, not daily starting every midnight. I created some additional 24-hour sensors myself using the HA statistics helper based off of the Total sensors, but I'm not certain they is accurate, in fact I'm pretty sure they are not, some of the numbers don't seem to add up.

What triggers binary_sensor.inverter_battery_alarm ? This sensor sometimes changes to "Problem" for a while. When I look at the official Solarman app there is no message, and the batteries themselves do not indicate any alert. What should I look for to solve this issue?

**Comment by davidrapan** [comment]
Hi @sfox38, most of the sensors are just values read from modbus registers of the inverter so what is shown depends on the inverter.

* When comes to the Grid sensor this is state of AC relay
* Battery Alarm can trigger problem even w/ simple 100% OVP (over-voltage protection which is BMS "standard" behavior)

Comparing data with Solarman App is not enough. Low update interval of 5 minutes can hide many things. That means any event w/ duration shorter than that doesn't have to be shown there.

But this frequent changing of Grid state is really not normal (I have the same device so believe me that the integration is very well tested for this model).

**So I would advise you to observe this events and states BUT compare it directly w/ the inverter using the display.**

👍
